homeassistant/components/adaptive_lighting/switch.py

```diff
@@ -0,0 +1,890 @@
+        sleep_mode_switch: AdaptiveSleepModeSwitch,
+    ):
+        """Initialize the Adaptive Lighting switch."""
+        self.hass = hass
+        self.turn_on_off_listener = turn_on_off_listener
+        self.sleep_mode_switch = sleep_mode_switch
+
+        data = validate(config_entry)
+        self._name = data[CONF_NAME]
+        self._lights = data[CONF_LIGHTS]
+
+        self._adapt_brightness = data[CONF_ADAPT_BRIGHTNESS]
+        self._adapt_color_temp = data[CONF_ADAPT_COLOR_TEMP]
+        self._adapt_rgb_color = data[CONF_ADAPT_RGB_COLOR]
+        self._initial_transition = data[CONF_INITIAL_TRANSITION]
+        self._interval = data[CONF_INTERVAL]
+        self._only_once = data[CONF_ONLY_ONCE]
+        self._prefer_rgb_color = data[CONF_PREFER_RGB_COLOR]
+        self._take_over_control = data[CONF_TAKE_OVER_CONTROL]
+        self._transition = data[CONF_TRANSITION]
+
+        self._sun_light_settings = SunLightSettings(
+            name=self._name,
+            astral_location=get_astral_location(self.hass),
+            max_brightness=data[CONF_MAX_BRIGHTNESS],
+            max_color_temp=data[CONF_MAX_COLOR_TEMP],
+            min_brightness=data[CONF_MIN_BRIGHTNESS],
+            min_color_temp=data[CONF_MIN_COLOR_TEMP],
+            sleep_brightness=data[CONF_SLEEP_BRIGHTNESS],
+            sleep_color_temp=data[CONF_SLEEP_COLOR_TEMP],
+            sunrise_offset=data[CONF_SUNRISE_OFFSET],
+            sunrise_time=data[CONF_SUNRISE_TIME],
+            sunset_offset=data[CONF_SUNSET_OFFSET],
+            sunset_time=data[CONF_SUNSET_TIME],
+            time_zone=self.hass.config.time_zone,
+        )
+
+        # Set other attributes
+        self._icon = ICON
+        self._entity_id = f"switch.{DOMAIN}_{slugify(self._name)}"
+        self._state = None
+
+        # Tracks 'off' → 'on' state changes
+        self._on_to_off_event: Dict[str, Event] = {}
+        # Tracks 'on' → 'off' state changes
+        self._off_to_on_event: Dict[str, Event] = {}
+        # Locks that prevent light adjusting when waiting for a light to 'turn_off'
+        self._locks: Dict[str, asyncio.Lock] = {}
+
+        # Set in self._update_attrs_and_maybe_adapt_lights
+        self._light_settings = {}
+
+        # Set and unset tracker in async_turn_on and async_turn_off
+        self.remove_listeners = []
+        _LOGGER.debug(
+            "%s: Setting up with '%s',"
+            " config_entry.data: '%s',"
+            " config_entry.options: '%s', converted to '%s'.",
+            self._name,
+            self._lights,
+            config_entry.data,
+            config_entry.options,
+            data,
+        )
+
+    @property
+    def entity_id(self):
+        """Return the entity ID of the switch."""
+        return self._entity_id
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return f"Adaptive Lighting: {self._name}"
+
+    # @property
```

Comment:
suddenly when adding this (the uncommented version), I got this error in my log:
```python
2020-10-02 15:39:49 ERROR (MainThread) [homeassistant.components.switch] Error while setting up adaptive_lighting platform for switch
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 192, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 301, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 405, in _async_add_entity
    entity.entity_id = entry.entity_id
AttributeError: can't set attribute
```

Any ideas why?

Reply:
You are forcing entity I'd. You should only be setting a default name and unique I'd. Then user can rename entity I'd and name from GUI.

Reply:
Thanks, that fixed it! 👍